apps/workspace-host/src/interface.ts

```diff
@@ -33,6 +33,43 @@ import { config, loadConfig } from './lib/config.js';
 import { parseDockerLogs } from './lib/docker.js';
 import { REPOSITORY_ROOT_PATH, APP_ROOT_PATH } from './lib/paths.js';
 import * as socketServer from './lib/socket-server.js';
+let server: http.Server | undefined;
+
+interface WorkspaceServerSettings {
+  instance_id?: string;
+  hostname?: string;
+  server_to_container_hostname?: string;
+  port?: number;
+}
+
+interface WorkspaceSettings {
+  workspace_image: string;
+  workspace_port: number;
+  workspace_home: string;
+  workspace_graded_files: string[];
+  workspace_args: string;
+  workspace_enable_networking: boolean;
+  workspace_environment: string[];
+}
+
+/*
+These types are somewhat duplicated from WorkspaceSchema in apps/prairielearn.
+TODO: break out types into a shared package.
+*/
```

Comment:
Nit, comments like this should always be formatted in JSDoc style.

```suggestion
/**
 * These types are somewhat duplicated from WorkspaceSchema in 
 * apps/prairielearn.
 * TODO: break out types into a shared package.
 */
```